# Project 2 -- [Ethan] [Wang]

**TA Help:** No one

- H
    
**Collaboration:** No one
    
- H

## Question 1

In [7]:
%%R
# We are reading the csv file into the myDF
myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# This displays the first few columns
head(myDF)

  ID                     Name Sex Age Height Weight           Team NOC
1  1                A Dijiang   M  24    180     80          China CHN
2  2                 A Lamusi   M  23    170     60          China CHN
3  3      Gunnar Nielsen Aaby   M  24     NA     NA        Denmark DEN
4  4     Edgar Lindenau Aabye   M  34     NA     NA Denmark/Sweden DEN
5  5 Christine Jacoba Aaftink   F  21    185     82    Netherlands NED
6  5 Christine Jacoba Aaftink   F  21    185     82    Netherlands NED
        Games Year Season      City         Sport
1 1992 Summer 1992 Summer Barcelona    Basketball
2 2012 Summer 2012 Summer    London          Judo
3 1920 Summer 1920 Summer Antwerpen      Football
4 1900 Summer 1900 Summer     Paris    Tug-Of-War
5 1988 Winter 1988 Winter   Calgary Speed Skating
6 1988 Winter 1988 Winter   Calgary Speed Skating
                               Event Medal
1        Basketball Men's Basketball  <NA>
2       Judo Men's Extra-Lightweight  <NA>
3            Football Me

In [8]:
%%R
# We are reading the csv file into the myDF
myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# returns each year as well as their athletes
table(myDF$Year)


 1896  1900  1904  1906  1908  1912  1920  1924  1928  1932  1936  1948  1952 
  380  1936  1301  1733  3101  4040  4292  5693  5574  3321  7401  7480  9358 
 1956  1960  1964  1968  1972  1976  1980  1984  1988  1992  1994  1996  1998 
 6434  9235  9480 10479 11959 10502  8937 11588 14676 16413  3160 13780  3605 
 2000  2002  2004  2006  2008  2010  2012  2014  2016 
13821  4109 13443  4382 13602  4402 12920  4891 13688 


In [ ]:
# Notes
# %%R
# # We are creating the table for the olympics data for athlete_events
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# myTable <- table(myDF$Year)

In [1]:
# Notes
# %%R
# # Here we are plotting the data for the olympics data for athlete_events
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# plot(myTable)

By using a plot, we can plot the data from the table and visualize the data that we have read into the data frame.


I began by printing the first six rows of the dataset to get a better understanding of the data. Following that, I printed the table as instructed. In this table, the first row in each alternating pair represents the year, while the next row shows the number of athletes participating that year. The years 1914, 1940, and 1944 did not have Olympic Games due to the world wars. Additionally, starting from 1992, the Olympics have alternated every four years between summer and winter events. The overall trend shows a general increase in the number of athletes, with significantly fewer athletes participating in the winter Olympics compared to the summer Olympics.

## Question 2

In [16]:
%%R
# The NOC column is now being sorted
sort(table(myDF$NOC))


  NFL   NBO   UNK   SSD   YMD   TUV   KOS   TLS   BRU   CRT   KIR   YAR   RHO 
    1     2     2     3     5     7     8     9    10    11    11    11    12 
  ROT   NRU   MHL   STP   CPV   COM   DMA   WIF   GBS   PLE   PLW   FSM   SOL 
   12    13    14    15    18    19    19    20    21    21    25    26    27 
  LCA   MTN   SOM   YEM   VAN   BHU   ASA   DJI   BDI   COK   GEQ   CHA   ARU 
   28    29    32    32    35    36    37    38    39    40    40    41    42 
  SKN   VIN   ERI   BUR   IVB   MAL   TGA   NIG   MDV   GRN   LAO   TKM   BAN 
   42    42    43    44    45    45    46    47    49    54    54    54    55 
  GAM   RWA   TOG   VNM   CAM   GAB   SAM   LES   OMA   CAF   SWZ   SUR   ALB 
   56    56    59    62    63    63    63    66    66    67    67    69    70 
  TJK   BEN   GUI   LBR   BIZ   NAM   MOZ   AHO   JOR   CAY   LBA   MKD   ANZ 
   70    71    73    74    76    77    78    79    80    83    84    84    86 
  MAW   BOT   MLI   COD   SAA   IOA   MNE   NEP   G

In [15]:
%%R
# Returns a couple of the highest values form the sorted table
tail(sort(table(myDF$Name)))


Jean Lucien Nicolas Jacoby                Takashi Ono 
                        32                         33 
       Ioannis Theofilakis      Joseph "Josy" Stoffel 
                        36                         38 
  Heikki Ilmari Savolainen       Robert Tait McKenzie 
                        39                         58 


In [10]:
# Notes
# %%R
# # Here we are doing a little bit of testing before we go ahead and view the entire list of authors because otherwise if the data is too much that would be bad
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# # Returns the head of the column "NOC"
# head(table(myDF$NOC))
# # Returns the tail of the column "NOC"
# tail(table(myDF$NOC))

In [11]:
# Notes
# %%R
# # Here we are doing a little bit of testing before we go ahead and view the entire list of authors because otherwise if the data is too much that would be bad
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# # Returns the head of the column "Name"
# head(table(myDF$Name))
# # Returns the tail of the column "Name"
# tail(table(myDF$Name))

In [68]:
# Notes
# %%R
# # Here we are creating another data frame but this time we are reading in the reviews for ice cream
# myDF <- read.csv("/anvil/projects/tdm/data/icecream/combined/reviews.csv")

In [67]:
# Notes
# %%R
# # This is just an example of how even though this is just the head, there is still a lot of data
# myDF <- read.csv("/anvil/projects/tdm/data/icecream/combined/reviews.csv")
# head(myDF)

In [69]:
# Notes
# %%R
# # Here we can make a table of the brands
# myDF <- read.csv("/anvil/projects/tdm/data/icecream/combined/reviews.csv")
# table(myDF$brand)

In [ ]:
# Notes
# %%R
# # Here we are doing a little bit of testing before we go ahead and view the entire list of authors because otherwise if the data is too much that would be bad
# myDF <- read.csv("/anvil/projects/tdm/data/icecream/combined/reviews.csv")
# length(table(myDF$author))

In [ ]:
# Notes
# %%R
# # Here we are viewing the tail of the table
# myDF <- read.csv("/anvil/projects/tdm/data/icecream/combined/reviews.csv")
# tail(table(myDF$author))

In [70]:
# Notes
# %%R
# # Here we are only viewing the head of the table
# myDF <- read.csv("/anvil/projects/tdm/data/icecream/combined/reviews.csv")
# head(table(myDF$author))

In [15]:
# Notes
# %%R
# # Here we are sorting the table before looking at the tail
# myDF <- read.csv("/anvil/projects/tdm/data/icecream/combined/reviews.csv")
# tail(sort(table(myDF$author)))

integer(0)


In [71]:
# Notes
# %%R
# # Here we are sorting the table before looking at the tail last 20
# myDF <- read.csv("/anvil/projects/tdm/data/icecream/combined/reviews.csv")
# tail(sort(table(myDF$author)), n=20)

You are able to choose whether you want to view the head of the table or the tail of the table which is like viewing the front few values and the back few values respectively. You can also sort the table and find the length of the table.

I initially used table(myDF$NOC) and noticed that the data was not sorted. To address this, I applied the sort() function. The USA appears the most frequently, with a count of 18,853. Since the list of names was extensive, I utilized the tail() function to focus on the entries with the highest occurrences, as we were interested in the most frequent entries, not the least (which would have required head()). Among the names, Robert Tait McKenzie appeared with a count of 58.


## Question 3

In [26]:
%%R
# Returns the first couple of values
head(myDF)

  ID                     Name Sex Age Height Weight           Team NOC
1  1                A Dijiang   M  24    180     80          China CHN
2  2                 A Lamusi   M  23    170     60          China CHN
3  3      Gunnar Nielsen Aaby   M  24     NA     NA        Denmark DEN
4  4     Edgar Lindenau Aabye   M  34     NA     NA Denmark/Sweden DEN
5  5 Christine Jacoba Aaftink   F  21    185     82    Netherlands NED
6  5 Christine Jacoba Aaftink   F  21    185     82    Netherlands NED
        Games Year Season      City         Sport
1 1992 Summer 1992 Summer Barcelona    Basketball
2 2012 Summer 2012 Summer    London          Judo
3 1920 Summer 1920 Summer Antwerpen      Football
4 1900 Summer 1900 Summer     Paris    Tug-Of-War
5 1988 Winter 1988 Winter   Calgary Speed Skating
6 1988 Winter 1988 Winter   Calgary Speed Skating
                               Event Medal
1        Basketball Men's Basketball  <NA>
2       Judo Men's Extra-Lightweight  <NA>
3            Football Me

In [27]:
%%R
# Returns the values that are identical to "Denmark"
table(myDF$Team)["Denmark"]

Denmark 
   3424 


In [29]:
%%R
# Returns the number of values in the table that have "Denmark" in it
length(grep("Denmark", myDF$Team))

[1] 3496


In [30]:
%%R
# Returns the number of values that have either identical or include the value "Denmark"
table(myDF$Team[ grepl("Denmark", myDF$Team) & (myDF$Team != "Denmark") ])


        Denmark-1         Denmark-2         Denmark-3         Denmark-4 
               30                30                 2                 2 
   Denmark/Sweden Miss Denmark 1964 
                6                 2 


In [19]:
# notes
# %%R
# # The number of rows corresponding exactly to team "Denmark".
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# sum(myDF$Team == "Denmark")

In [18]:
# notes
# %%R
# # The number of rows with "Denmark" as part of the team name.
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# sum(myDF$Team != "Denmark")

In [17]:
# notes
# %%R
# # The names of teams that have "Denmark" included but are not exactly "Denmark".
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# myDF$Team[ grepl("Denmark", myDF$Team) & (myDF$Team != "Denmark") ]

In [20]:
# notes
# %%R
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# head(myDF)
# # Test if a value is equal to japan
# tail(myDF$Team == "Japan")
# # R turns FALSE value to 0, and TRUE values to 1,
# # so we can find th enumber of TRUE values by just summing them up
# sum(myDF$Team == "Japan")

In [21]:
# notes
# %%R
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# # Test if a value is equal to United States
# sum(myDF$Team == "United States")
# # Make a table with only the united states
# table(myDF$Team)["United States"]

In [22]:
# notes
# %%R
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# sum(myDF$Team == "United States")
# table(myDF$Team)["United States"]
# # two tools to search for patterns
# # grep finds the locations where a pattern occurs
# # grepl returns TRUE or FALSE according to whether or not the pattern occurs
# head(grep("Japan", myDF$Team))
# # You can also find the length
# length(grep("Japan", myDF$Team))

In [61]:
# notes
# %%R
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# # Shows how many are TRUE or FALSE
# head(grepl("Japan", myDF$Team))
# # when we sum TRUE and FALSE values,
# # the FALSE values become 0 and the TRUE values become 1# so the sum is the number of TRUE values
# sum(grepl("Japan", myDF$Team))
# myDF$Team[ grepl("Japan", myDF$Team) & (myDF$Team != "Japan") ]
# # remember: 8289 teams have *exactly* the team name "Japan"
# # and 8423 teams have "Japan" as part of the name, but maybe not the whole name
# # what are the 8423 - 8289 = 134 team names
# # that have Japan but are not exactly Japan?

In [23]:
# notes
# %%R
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# # Makes this into a table
# table(myDF$Team[ grepl("Japan", myDF$Team) & (myDF$Team != "Japan") ])

In [ ]:
# notes
# %%R
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# # Figure out the number of something
# table(myDF$Team)["Germany"]

In [24]:
# notes
# %%R
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# # sum the number of TRUE values
# sum(myDF$Team == "Germany")

In [25]:
# notes
# %%R
# myDF <- read.csv("/anvil/projects/tdm/data/olympics/athlete_events.csv")
# # sum the number of Germany
# length(grep("Germany", myDF$Team))
# # makes TRUE and FALSE values, 
# # according to whether or not "Germany" is in the team name
# # and then we can sum up the TRUE values to see how many there are
# # remember: when we sum, TRUE becomes 1 and FALSE becomes 0
# #grep("Germany", myDF$Team)

There are exactly 3,424 rows that correspond specifically to “Denmark.” Additionally, there are 3,496 rows that include team names containing “Denmark.” The names of these teams are “Denmark-1,” “Denmark-2,” “Denmark-3,” “Denmark-4,” “Denmark/Sweden,” and “Miss Denmark 1964.”


## Question 4

In [31]:
%%R
# This will import the library that is data.table that allows us to use fread which reads a csv file and separates words based on an indicator value
library(data.table)
myDF <- fread("/anvil/projects/tdm/data/election/itcont1980.txt", quote="")
names(myDF) <- c("CMTE_ID", "AMNDT_IND", "RPT_TP", "TRANSACTION_PGI", "IMAGE_NUM", "TRANSACTION_TP", "ENTITY_TP", "NAME", "CITY", "STATE", "ZIP_CODE", "EMPLOYER", "OCCUPATION", "TRANSACTION_DT", "TRANSACTION_AMT", "OTHER_ID", "TRAN_ID", "FILE_NUM", "MEMO_CD", "MEMO_TEXT", "SUB_ID")

data.table 1.15.4 using 1 threads (see ?getDTthreads).  Latest news: r-datatable.com


In [34]:
%%R
# Outputs the values at the highest from the sorted list of the NAME column
tail(sort(table(myDF$NAME)))


                            LUCAS, HARRY JR 
                                         46 
                              WILLIAMS, J D 
                                         64 
  DEMOCRATIC CONGRESSIONAL DINNER COMMITTEE 
                                         65 
                               REPP, WARD W 
                                         89 
 AMERICAN DENTAL POLITICAL ACTION COMMITTEE 
                                         96 
AMERICAN MEDICAL POLITICAL ACTION COMMITTEE 
                                        769 


In [35]:
%%R
# Outputs the values at the highest from the sorted list of the STATE column
tail(sort(table(myDF$STATE)))


         FL    IL    NY    TX    CA 
12654 15576 15895 25669 34862 37069 


In [36]:
# notes
# %%R
# # load up data.table library
# library(data.table)
# # Because we have the data.table library we are able to use fread
# myDF <- fread("/anvil/projects/tdm/data/election/itcont1980.txt", quote="")
# names(myDF) <- c("CMTE_ID", "AMNDT_IND", "RPT_TP", "TRANSACTION_PGI", "IMAGE_NUM", "TRANSACTION_TP", "ENTITY_TP", "NAME", "CITY", "STATE", "ZIP_CODE", "EMPLOYER", "OCCUPATION", "TRANSACTION_DT", "TRANSACTION_AMT", "OTHER_ID", "TRAN_ID", "FILE_NUM", "MEMO_CD", "MEMO_TEXT", "SUB_ID")
# # Returns the head of the column "STATE"
# head(table(myDF$STATE))
# # Returns the tail of the column "STATE"
# tail(table(myDF$STATE))

In [33]:
# notes
# %%R
# # load up data.table library
# library(data.table)
# # Because we have the data.table library we are able to use fread
# myDF <- fread("/anvil/projects/tdm/data/election/itcont1980.txt", quote="")
# names(myDF) <- c("CMTE_ID", "AMNDT_IND", "RPT_TP", "TRANSACTION_PGI", "IMAGE_NUM", "TRANSACTION_TP", "ENTITY_TP", "NAME", "CITY", "STATE", "ZIP_CODE", "EMPLOYER", "OCCUPATION", "TRANSACTION_DT", "TRANSACTION_AMT", "OTHER_ID", "TRAN_ID", "FILE_NUM", "MEMO_CD", "MEMO_TEXT", "SUB_ID")
# # Returns the head of the column "NAME"
# head(table(myDF$NAME))
# # Returns the tail of the column "NAME"
# tail(table(myDF$NAME))

The most frequently appearing name is the American Medical Political Action Committee, with 769 occurrences. The state that appears the most is California (CA), with 37,069 occurrences.


## Question 5

In [38]:
%%R
# This will combine the city and state, then output the table of this combination process, sort it from lowest to highest, and return a couple of city and state combinations at the end
tail(sort(table(paste(myDF$CITY, myDF$STATE))))

SyntaxError: invalid syntax (3991573219.py, line 2)

The top 5 city-and-state locations for donations are New York, New York with 13,862 donations; Houston, Texas with 10,146 donations; Dallas, Texas with 6,438 donations; Washington, DC with 5,890 donations; and Los Angeles, California with 5,866 donations.

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.